### ISSUE TITLE
[java] UnnecessaryImport FP with static imports of overloaded methods

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#4557]
**Affects PMD Version:** 7.0.0-rc1/rc2

**Rule:** [UnnecessaryImport](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryimport)

**Description:**

I've noticed this via #4501 and thought it would be the same as #4432, but it's actually a different case. I've now been able to reproduce it with a minimal example:

**Code Sample demonstrating the issue:**

File `aaa/Truth.java`:
```java
package aaa;

public final class Truth {
    public static Object assertThat(Integer actual) {
        return null;
    }

    public static Object assertThat(Long actual) {
        return null;
    }
}
```

File `bbb/Sample.java`:
```java
package bbb;

import static aaa.Truth.assertThat; // false positive violation here

public class Sample {
    public static void main(String[] args) {
        assertThat(1); // usage of aaa.Truth::assertThat(Integer)
    }
}
```

**Expected outcome:**

PMD reports a violation in bbb/Sample.java at line 3, but that's wrong. That's a false positive.
It's used in line 7.

**More info:**

This is something with overload resolution: We detect that the two static methods could be used, but we think, this is ambiguous - and that's why we fail the resolution:

https://github.com/pmd/pmd/blob/3a9c3616829221e22a5de78ce07c1e2cc394b02c/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/types/internal/infer/PhaseOverloadSet.java#L84
